ipa-core/src/protocol/ipa_prf/validation_protocol/proof_generation.rs

```diff
@@ -0,0 +1,475 @@
+    /// it iteratively generates proofs and collects them in a batch
+    ///
+    /// Outputs `(my_batch_left_shares, shares_of_batch_from_prover_left, (p_mask_from_left_prover, q_mask_from_right_prover))`
+    ///
+    /// ## Panics
+    /// Panics when masks cannot be set safely (implementation issue).
+    fn generate_batch_parts<C, I>(
+        ctx: &C,
+        uv_tuple_inputs: I,
+    ) -> (Self, Self, (Fp61BitPrime, Fp61BitPrime))
+    where
+        C: Context,
+        I: Iterator<Item = UVTupleBlock<Fp61BitPrime>> + Clone,
+    {
+        // set up record counter
+        let mut record_counter = RecordId::from(0);
+
+        // generate first proof from input iterator
+        let (mut uv_values, first_proof_from_left, prover_left_first_proof) = {
+            // precomputation for first proof
+            // first denominator
+            let first_denominator = CanonicalLagrangeDenominator::<
+                Fp61BitPrime,
+                { LargeProofGenerator::RECURSION_FACTOR },
+            >::new();
+            // first lagrange table
+            let first_lagrange_table = LagrangeTable::<
+                Fp61BitPrime,
+                { LargeProofGenerator::RECURSION_FACTOR },
+                { LargeProofGenerator::LAGRANGE_LENGTH },
+            >::from(first_denominator);
+
+            LargeProofGenerator::gen_artefacts_from_recursive_step(
+                ctx,
+                &mut record_counter,
+                &first_lagrange_table,
+                Self::polynomials_from_inputs(uv_tuple_inputs),
+            )
+        };
+
+        // storage for other proofs
+        let mut my_proofs_left_shares =
+            Vec::<[Fp61BitPrime; SmallProofGenerator::PROOF_LENGTH]>::new();
+        let mut shares_of_proofs_from_prover_left =
+            Vec::<[Fp61BitPrime; SmallProofGenerator::PROOF_LENGTH]>::new();
+        // generate masks
+        // verifier on the right has p,
+        // therefore the right share is "implicitly sent" to the right ("communicated" via PRSS)
+        let (p_mask_from_left_prover, my_p_mask): (Fp61BitPrime, Fp61BitPrime) =
+            ctx.prss().generate_fields(record_counter);
+        record_counter += 1;
+        // and verifier on the left has q
+        // therefore the left share is "implicitly sent" to the left (communication via PRSS)
+        let (my_q_mask, q_mask_from_right_prover): (Fp61BitPrime, Fp61BitPrime) =
+            ctx.prss().generate_fields(record_counter);
+        record_counter += 1;
+
+        // precomputation for other proofs
+        // denominator
+        let denominator = CanonicalLagrangeDenominator::<
+            Fp61BitPrime,
+            { SmallProofGenerator::RECURSION_FACTOR },
+        >::new();
+        // lagrange table
+        let lagrange_table = LagrangeTable::<
+            Fp61BitPrime,
+            { SmallProofGenerator::RECURSION_FACTOR },
+            { SmallProofGenerator::LAGRANGE_LENGTH },
```

Comment:
This code takes up a LOT more vertical space and becomes much less readable due to the size of these generic params. Can you assign these to a constant with a short name like `X` or something and avoid all of this wrapping?